qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/RolesAndPrivilegesSteps.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import static org.junit.Assert.assertTrue;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.page.AddEditNewPrivilegePage;
+import org.openmrs.contrib.qaframework.page.AddNewRolePage;
+import org.openmrs.contrib.qaframework.page.AdministrationManageRolesPage;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.ConfigureMetadataPage;
+import org.openmrs.contrib.qaframework.page.ManagePrivilegesPage;
+import org.openmrs.contrib.qaframework.page.SystemAdministrationPage;
+
+public class RolesAndPrivilegesSteps extends Steps {
+
+	private static final String ROLE_DESCRIPTION = "for e2e automation test";
+	private static final String PRIVILEGE_DESCRIPTION = "This privilege is just one developed for the roles&privileges e2e test";
+	private ManagePrivilegesPage manageprivilegesPage;
+	private ConfigureMetadataPage configuremetadatapage;
+	private AddEditNewPrivilegePage addNewPrivilegePage;
+	private AddNewRolePage addNewRolePage;
+	private SystemAdministrationPage systemAdministrationPage;
+	private AdministrationPage administrationPage;
+	private AdministrationManageRolesPage administrationManageRolesPage;
+	String privilegeName =  RandomStringUtils.randomAlphabetic(10);
+	String roleName =  RandomStringUtils.randomAlphabetic(10);
```

Comment:
This is a good approach! One comment though; aren't these variables supposed to be final?